librz/util/float/float.c

```diff
@@ -1,31 +1,92 @@
+// SPDX-FileCopyrightText: 2022 heersin <teablearcher@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+/**
+ * \file float.c
+ * This file implements IEEE-754 binary float number operations (32/64/128)
+ * IEEE binary representations, use binary digits to represent float. machine-friendly
+ * binary32 format (single) : use a 32 bits bitvector to represent float
+ * 	32 bits = 1 (sign bit) + 8 (exponent bits) + 23 (mantissa bits)
+ * 	exponent value range : -126 ~ 127
+ * binary64 format (double) : use a 64 bits bitvector to represent float
+ *  	64 bits = 1 (sign bit) + 11 (exponent bits) + 52 (mantissa bits)
+ * 	exponent value range : -1022 ~ 1023
+ * binary128 format, use a 128 bits bitvector to represent float
+ *  	128 bits = 1 (sign bit) + 15 (exponent bits) + 112 (mantissa bits)
+ * 	exponent value range : -16382 ~ 16383
+ **/
+
 #include "float_internal.c"
 
 /**
- * This file implements IEEE-754 binary float number (32/64/128)
+ * \breif return the bitvector string of a float
+ * \param f float
+ * \return char* string of bitvector
+ */
+RZ_API RZ_OWN char *rz_float_bv_as_string(RZ_NULLABLE RzFloat *f) {
+	if (!f || !f->s) {
+		return strdup("NULL");
```

Comment:
Maybe it should be `NULL` pointer instead of the string `"NULL"`

Reply:
i agree.